crates/bevy_ui/src/layout/ui_surface.rs

```diff
@@ -277,11 +277,21 @@ impl UiSurface {
     /// Does not compute the layout geometry, `compute_window_layouts` should be run before using this function.
     /// On success returns a pair consisting of the final resolved layout values after rounding
     /// and the size of the node after layout resolution but before rounding.
-    pub fn get_layout(&mut self, entity: Entity) -> Result<(taffy::Layout, Vec2), LayoutError> {
+    pub fn get_layout(
+        &mut self,
+        entity: Entity,
+        use_rounding: bool,
+    ) -> Result<(taffy::Layout, Vec2), LayoutError> {
         let Some(taffy_node) = self.entity_to_taffy.get(&entity) else {
             return Err(LayoutError::InvalidHierarchy);
         };
 
+        if use_rounding {
+            self.taffy.enable_rounding();
+        } else {
+            self.taffy.disable_rounding();
+        }
```

Comment:
This is potentially leaky if the `self.taffy.layout(..)` errors-out. It seems this function should always end with rounding enabled.

Reply:
Yeah this function is fine I think as this is the only place we read the layout values and we reset the rounding mode each time we enter it, but we need to call `enable_rounding()` each frame before we recompute the layout since like you say the update walk could end with `disable_rounding` set.

Reply:
> Yeah this function is fine I think as this is the only place we read the layout values and we reset the rounding mode each time we enter it

Imo this reasoning is too fragile. It assumes the current design will stay the same, and it implies if the design does change then whoever changes it will somehow know the right thing to do.